client/reducers/scenario.js

```diff
@@ -11,21 +25,32 @@ const initialState = {
     status: 1
 };
 
-export default function(state = initialState, action) {
-    switch (action.type) {
-        case SET_SCENARIO:
-        case SET_SCENARIOS: {
-            return {
-                ...state,
-                ...action.payload
-            };
-        }
-        case SET_SLIDES:
-            return {
-                ...state,
-                ...action.payload
-            };
-        default:
-            return state;
+export const scenario = (state = initialState, action) => {
+    const { scenario, slides, type } = action;
+
+    if (type === SET_SCENARIO || type === GET_SCENARIO_SUCCESS) {
```

Comment:
non-blocking nit: a switch statement might be easier to read rather than two if-statements.

Reply:
Yeah, that's true. I could do a cleanup in a follow up :)